File: Data Structures & Algorithms/max-consecutive-ones/submission-2.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxConsecutive = 0;
            int size = nums.size();

            for (int i=0; i < size; i++) {

                if (nums[i] == 1) {

                    int j = 0;

                    do {
                        j++;
                        i+=1;
                    } while (i != size && nums[i] == 1);

                    maxConsecutive = j > maxConsecutive? j : maxConsecutive;
                }
            }
            return maxConsecutive;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int[] nums = new int[2];

        int p1 = 0;
        int p2 = numbers.Length - 1;
        int sum = numbers[p1] + numbers[p2];

        do 
        {
            if (sum > target)
                p2--;
            else if (sum < target)
                p1++;
            else if (sum == target)
            {
                nums[0] = p1 + 1;
                nums[1] = p2 + 1;
                return nums;
            }

            sum = numbers[p1] + numbers[p2];
        }
        while (p1 != p2);

        return nums;
    }
}
